src/www/ui/page/AdviceLicense.php

```diff
@@ -163,19 +174,12 @@ private function saveInput(Request $request, $oldRow)
       throw new \Exception('shortname already in use');
     }
     if ($oldRow['rf_pk'] == -1) {
-      $oldRow['rf_pk'] = $licenseDao->insertUploadLicense($shortname, $rfText, Auth::getGroupId());
+      $oldRow['rf_pk'] = $licenseDao->insertUploadLicense($shortname, $rfText, Auth::getGroupId(), $userId);
     }
 
     $licenseDao->updateCandidate($oldRow['rf_pk'], $shortname, $fullname,
-      $rfText, $url, $note, !empty($marydone), $riskLvl);
-    return array('rf_pk' => $oldRow['rf_pk'],
-        'rf_shortname' => $shortname,
-        'rf_fullname' => $fullname,
-        'rf_text' => $rfText,
-        'rf_url' => $url,
-        'rf_notes' => $note,
-        'rf_risk' => $riskLvl,
-        'marydone' => $marydone);
+      $rfText, $url, $note, $lastmodified, $userIdmodified, !empty($marydone), $riskLvl);
+      return $this->getDataRow(Auth::getGroupId(), $oldRow['rf_pk']);
```

Comment:
Please fix the indentation.

```suggestion
    return $this->getDataRow(Auth::getGroupId(), $oldRow['rf_pk']);
```

Reply:
done